jgrapht-core/src/main/java/org/jgrapht/intervalgraph/BinarySearchTree.java

```diff
@@ -0,0 +1,182 @@
+package org.jgrapht.intervalgraph;
+
+/**
+ * Interface for Binary Search Trees
+ *
+ * @param <K> the key
+ * @param <V> the value
+ *
+ * @author Christoph Grüne (christophgruene)
+ * @since Apr 18, 2018
+ */
+public interface BinarySearchTree<K, V> {
```

Comment:
All of these classes will be moved to the util package. Maybe one wants to reuse it. Either way, it has to be public. We left everything in one package, because we wanted to have an overview. In the future it will be helpful to have this as well. At the end it will be moved to the corresponding packages.